extensions/lifecycle/LifecycleOplogPopulatorUtils.js

```diff
@@ -1,27 +1,59 @@
 const OplogPopulatorUtils = require('../../lib/util/OplogPopulatorUtils');
+const locations = require('../../conf/locationConfig.json');
 
 class LifecycleOplogPopulatorUtils extends OplogPopulatorUtils {
+
     /**
      * Get extension specific MongoDB filter
      * to get buckets that have the lifecycle
      * extension enabled
      * @returns {Object} MongoDB filter
      */
      static getExtensionMongoDBFilter() {
-        // getting buckets with at least one lifecycle
-        // rule enabled
+        const coldLocations = Object.keys(locations).filter(loc => locations[loc].isCold);
+        // getting buckets with at least one lifecycle transition rule enabled
+        // or buckets with disabled lifecycle transition rules to cold locations (needed for restores)
         return {
             'value.lifecycleConfiguration.rules': {
                 $elemMatch: {
-                    ruleStatus: 'Enabled',
-                    actions: {
-                        $elemMatch: {
-                            $or: [
-                                { actionName: 'Transition' },
-                                { actionName: 'NoncurrentVersionTransition' },
-                            ],
+                    $or: [
```

Comment:
not sure what is the most "efficient" (for mongo), or simple: could/shoud we check the following instead?

```
$elemMatch: {
   $and: [
       actions: {
          $elemMatch: {
              actionName: {
                  $in: ['Transition', 'NoncurrentVersionTransition'],
              },
          },
       },
      $or: [
         { ruleStatus: 'Enabled' },
         { actions: {
		transition: {
                    $elemMatch: { storageClass: { $in: coldLocations }},
                 },
         }},
         { actions: {
		nonCurrentVersionTransition: {
                    $elemMatch: { storageClass: { $in: coldLocations }},
                 },
         }},
      ]
   ]
}
```

Reply:
(seems conceptually simple, with less redundancy ; but not sure if it performs better or is more readable when finalized...)